routes/hot_feed.go

```diff
@@ -750,10 +788,21 @@ func (fes *APIServer) GetHotnessScoreInfoForTxn(
 		InteractionPKID:     *interactionPKIDEntry.PKID,
 		InteractionPostHash: *interactionPostHash,
 	}
-	if _, exists := postInteractionMap[interactionKey]; exists {
-		return nil, nil, 0
+
+	// Transaction type multiplier for current transaction.
+	multiplier := fes.getTxnMultiplier(txn)
+
+	// Get previously applied multiplier for post, if post has been counted already for this user.
+	if prevMultiplier, exists := postInteractionMap[interactionKey]; exists {
+		// If the previously applied multiplier is greater, skip this transaction.
+		if prevMultiplier > multiplier {
+			return nil, nil, 0
+		}
+		postInteractionMap[interactionKey] = multiplier
+		// We want to count the difference of the new multiplier and the previously counted multiplier.
+		multiplier = multiplier - prevMultiplier
```

Comment:
we're taking the diff here because this is only for the incremental multiplier achieved by this transaction right?

Reply:
Correct